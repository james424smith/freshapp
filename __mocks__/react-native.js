import * as ReactNative from "react-native";

export * from "react-native";
export const Vibration = { vibrate: jest.fn() };
export const Adnimated = { spring: jest.fn() };

export default Object.setPrototypeOf(
  {
    Vibration,
    Adnimated,
  },
  ReactNative
);
