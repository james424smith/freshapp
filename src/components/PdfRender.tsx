import React from "react";
import { View } from "react-native";
import { PdfResourceType } from "../interfaces";
import PDFView from "react-native-view-pdf";
import styles from "../styles/more_screens/PdfRenderStyles";

type Props = { document: string; resourceType: PdfResourceType };

const PdfRender = (props: Props) => {
  const { resourceType, document } = props;
  return (
    <View style={styles.root}>
      <PDFView
        fadeInDuration={10.0}
        style={styles.pdf}
        resource={document}
        resourceType={resourceType}
      />
    </View>
  );
};

export default PdfRender;
