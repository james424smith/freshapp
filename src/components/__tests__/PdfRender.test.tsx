import React from "react";
import PdfRender from "../PdfRender";
import { render, cleanup } from "@testing-library/react-native";

describe("PdfRender snapshot test", () => {
  afterEach(() => {
    cleanup();
    jest.clearAllMocks();
    jest.restoreAllMocks();
  });

  it("should render correct ui", () => {
    const { toJSON } = render(
      <PdfRender document={"1234"} resourceType={"base64"} />
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
