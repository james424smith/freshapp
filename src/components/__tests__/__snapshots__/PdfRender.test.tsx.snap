// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PdfRender snapshot test should render correct ui 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <PDFView
    fadeInDuration={10}
    onError={[Function]}
    onLoad={[Function]}
    onPageChanged={[Function]}
    onScrolled={[Function]}
    resource="1234"
    resourceType="base64"
    style={
      Object {
        "flex": 1,
        "width": 750,
      }
    }
    textEncoding="utf-8"
    urlProps={Object {}}
  />
</View>
`;
